airbyte-integrations/connectors/source-amazon-ads/source_amazon_ads/source.py

```diff
@@ -0,0 +1,114 @@
+#
+# MIT License
+#
+# Copyright (c) 2020 Airbyte
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+#
+
+
+from typing import Any, List, Mapping, Tuple
+
+from airbyte_cdk.logger import AirbyteLogger
+from airbyte_cdk.models import ConnectorSpecification
+from airbyte_cdk.sources import AbstractSource
+from airbyte_cdk.sources.streams import Stream
+from airbyte_cdk.sources.streams.http.auth import Oauth2Authenticator
+
+from .common import SourceContext
+from .spec import Spec
+from .streams import (
+    DisplayReportStream,
+    Profiles,
+    SponsoredBrandsAdGroups,
+    SponsoredBrandsCampaigns,
+    SponsoredBrandsKeywords,
+    SponsoredBrandsReportStream,
+    SponsoredDisplayAdGroups,
+    SponsoredDisplayCampaigns,
+    SponsoredDisplayProductAds,
+    SponsoredDisplayTargetings,
+    SponsoredProductAdGroups,
+    SponsoredProductAds,
+    SponsoredProductCampaigns,
+    SponsoredProductKeywords,
+    SponsoredProductNegativeKeywords,
+    SponsoredProductsReportStream,
+    SponsoredProductTargetings,
+)
+
+TOKEN_URL = "https://api.amazon.com/auth/o2/token"
+
+
+# Source
+class SourceAmazonAds(AbstractSource):
+    def __init__(self):
+        super().__init__()
+        self.ctx = SourceContext()
+
+    def check_connection(self, logger: AirbyteLogger, config: Mapping[str, Any]) -> Tuple[bool, any]:
+        """
+        :param config:  the user-input config object conforming to the connector's spec.json
+        :param logger:  logger object
+        :return Tuple[bool, any]: (True, None) if the input config can be used to connect to the API successfully, (False, error) otherwise.
+        """
+        config = Spec(**config)
```

Comment:
you see, there is a use for Config, or ConnectorConfig, but Spec is something artificial for connectors internals, it is used only in spec method. I think it is better to keep only config there and populate ConnectorSpecification in spec method. WDYT?

Reply:
ok